doc/DEVELOPER.adoc

```diff
@@ -109,6 +109,22 @@ To lock the package versions to ensure that they are not incidentially updated b
 
 === Testing the application
 
+Testing can be performed in the following modes
+
+Testing with NPM:: cf. <<Testing with NPM>>.
+
+NOTE:
+Both of the following modes are based on this mode and effectively call `npm` but are used via Maven which enables a local installation of the desired NPM and Node versions first.
+It then uses these versions to perform the build and test of the application.
+On CI (https://dev.dukecon.org/jenkins/job/dukecon_pwa/[The DukeCon Jenkins]) the build and tests are performed via this mode.
+
+Build with Maven (directly):: cf. <<Testing with Maven (direct)>>
```

Comment:
instead of referring to a section by name, a best practice it to give referenced sections an ID. This avoid breaking references accidentally when changing the title of a section. Applies in multiple places in the document.

Reply:
Resolved this by the latest commit on this branch.